packages/core/src/config/utils.ts

```diff
@@ -157,6 +157,7 @@ export function getMergedConfig(config: Config, apiName?: string): Config {
             ...config['features.mockServer'],
             ...config.apis[apiName]?.['features.mockServer'],
           },
+          files: [...new Set([...config.files, ...(config.apis?.[apiName]?.files ?? [])])],
```

Comment:
I'd place the uniqueness check  somewhere here:  https://github.com/Redocly/redocly-cli/pull/842/files#diff-51105770da26478a3f6a7c2ee8107952064a25f515f456f57b7a626ad5f9587dR269 to check all the uploaded files.